src/Result.php

```diff
@@ -153,6 +153,28 @@ protected function printSuccess()
         $this->printTaskSuccess("Done $time", $this->task);
     }
 
+    /**
+     * Merge the provided data array into this result.
+     * In the case of conflicting data keys, data already
+     * in this object takes precedence.
+     */
+    public function overlay($data)
+    {
+        $this->data = $data + $this->data;
+        return $this;
+    }
+
+    /**
+     * Merge another result into this result.  Data already
+     * existing in this result takes precedence over the
+     * data in the Result being merged.
+     */
+    public function merge(Result $result)
```

Comment:
I don't like mode-altering parameters; however, I think that `overlay` can probably go away.